lib/measured/measurable.rb

```diff
@@ -51,14 +51,20 @@ def <=>(other)
     if other.is_a?(self.class)
       other_converted = other.convert_to(unit)
       value <=> other_converted.value
+    elsif other == 0
+      other_converted = self.class.new(0, unit)
+      value <=> other_converted.value
     end
   end
 
   def ==(other)
-    return false unless other.is_a?(self.class)
-
-    other_converted = other.convert_to(unit)
-    value == other_converted.value
+    if other.is_a?(self.class)
+      other_converted = other.convert_to(unit)
+      value == other_converted.value
+    elsif other == 0
+      other_converted = self.class.new(0, unit)
+      value == other_converted.value
```

Comment:
I guess this is slightly more correct than `value == 0` but I can't think of even a future scenario where this would act differently.